adaptive/src/main/java/com/google/accompanist/adaptive/TwoPane.kt

```diff
@@ -0,0 +1,545 @@
+
+        layout(constraints.maxWidth, constraints.maxHeight) {
+            val splitResult = strategy.calculateSplitResult(
+                density = density,
+                layoutDirection = layoutDirection,
+                layoutCoordinates = checkNotNull(coordinates) {
+                    "TwoPane does not support the use of alignment lines!"
+                }
+            )
+
+            val isHorizontalSplit = splitResult.isHorizontalSplit
+            val gapBounds = splitResult.gapBounds
+
+            val gapLeft = constraints.constrainWidth(gapBounds.left.roundToInt())
+            val gapRight = constraints.constrainWidth(gapBounds.right.roundToInt())
+            val gapTop = constraints.constrainHeight(gapBounds.top.roundToInt())
+            val gapBottom = constraints.constrainHeight(gapBounds.bottom.roundToInt())
+            val firstConstraints =
+                if (isHorizontalSplit) {
+                    val width = when (layoutDirection) {
+                        LayoutDirection.Ltr -> gapLeft
+                        LayoutDirection.Rtl -> constraints.maxWidth - gapRight
+                    }
+
+                    constraints.copy(minWidth = width, maxWidth = width)
+                } else {
+                    constraints.copy(minHeight = gapTop, maxHeight = gapTop)
+                }
+            val secondConstraints =
+                if (isHorizontalSplit) {
+                    val width = when (layoutDirection) {
+                        LayoutDirection.Ltr -> constraints.maxWidth - gapRight
+                        LayoutDirection.Rtl -> gapLeft
+                    }
+                    constraints.copy(minWidth = width, maxWidth = width)
+                } else {
+                    val height = constraints.maxHeight - gapBottom
+                    constraints.copy(minHeight = height, maxHeight = height)
+                }
+            val firstPlaceable = firstMeasurable.measure(constraints.constrain(firstConstraints))
+            val secondPlaceable = secondMeasurable.measure(constraints.constrain(secondConstraints))
+
+            firstPlaceable.placeRelative(0, 0)
+            val detailOffsetX =
+                if (isHorizontalSplit) {
+                    constraints.maxWidth - secondPlaceable.width
+                } else {
+                    0
+                }
+            val detailOffsetY =
+                if (isHorizontalSplit) {
+                    0
+                } else {
+                    constraints.maxHeight - secondPlaceable.height
+                }
+            secondPlaceable.placeRelative(detailOffsetX, detailOffsetY)
+        }
+    }
+}
+
+/**
+ * Returns the specification for where to place a split in [TwoPane] as a result of
+ * [TwoPaneStrategy.calculateSplitResult]
+ */
+public class SplitResult(
+
+    /**
+     * Whether the split is vertical or horizontal
+     */
+    public val isHorizontalSplit: Boolean,
```

Comment:
Now if we steer away from the noun `split`, we have to rephrase it. Perhaps, `splitOrientation: Orientation` might be a good idea?

Reply:
Updated! I went with `gapOrientation`, and used the `Orientation` that already exists in Compose (might want to have a custom one?)

Reply:
Nah, I think the Orientation we have covers exactly what we want